test/tools/cli/IonProcess.java

```diff
@@ -265,115 +266,111 @@ private static void processToEventFromIonStream(IonWriter ionWriterForOutput,
-                    writeIonByType(event, ionWriterForOutput);
-                } else if (event.getEventType().equals(EventType.SYMBOL_TABLE)) {
-                    //TODO symbol table
-                    // For SYMBOL_TABLE events, flush the writer's existing local symbol table and any buffered data,
-                    // forcing the writer to create a new local symbol table that imports the list of symbol tables
-                    // declared by the imports field of the Event. This ensures that symbol tokens with unknown text
-                    // that occur in subsequent events in the stream can be written correctly
-                } else if (event.getEventType().equals(EventType.STREAM_END)) {
-                    ionWriterForOutput.finish();
+                                                       IonWriter ionWriterForErrorReport,
+                                                       IonReader ionReader,
+                                                       ProcessContext processContext,
+                                                       ProcessArgs args,
+                                                       OutputStream outputStream) throws IOException {
+        while (ionReader.next() != null) {
+            //update eventIndex
+            processContext.setEventIndex(processContext.getEventIndex() + 1);
+            Event event = eventStreamToEvent(ionReader);
+            //event.validate();
+            processContext.setLastEventType(event.getEventType());
+            if (event.getEventType() == EventType.CONTAINER_START) {
+
+                if (isEmbeddedEvent(event)) {
+                    embeddedEventToIon(ionReader, ionWriterForOutput, processContext, args, outputStream);
+                } else {
+                    IonType type = event.getIonType();
+                    setAnnotationAndField(event, ionWriterForOutput);
+                    ionWriterForOutput.stepIn(type);
                 }
+            } else if (event.getEventType().equals(EventType.CONTAINER_END)) {
+                ionWriterForOutput.stepOut();
+            } else if (event.getEventType().equals(EventType.SCALAR)) {
+                writeIonByType(event, ionWriterForOutput);
+            } else if (event.getEventType().equals(EventType.SYMBOL_TABLE)) {
+                ionWriterForOutput.finish();
+                ImportDescriptor[] imports = event.getImports();
+                SymbolTable[] symbolTables = new SymbolTable[imports.length];
+                for (int i = 0; i < imports.length; i++) {
+                    SymbolTable symbolTable = IonSystemBuilder.standard().build().newSharedSymbolTable(
+                            imports[i].getImportName(), imports[i].getVersion(), null);
+                    symbolTables[i] = symbolTable;
+                }
+                ionWriterForOutput = args.getOutputFormat().createIonWriterWithImports(outputStream, symbolTables);
```

Comment:
This reassignment does not last beyond the return of this method. You probably need to add the IonWriter to the ProcessContext so you can reassign it there and keep the new writer in scope for data written after the return of this method.

Reply:
Changed.

I Create a ionWriter at IonProcess.java 76 and close it at IonProcess.java 81-90

I did same thing for embedded Stream since we may face symbol table within an embedded stream.
(I create the processContext for embeddedStream at Ionprocess.java 340 and close it at line 375.)